app/utils.py

```diff
@@ -193,3 +202,72 @@ def remove_title_date(title: str) -> str:
     The date format expected is: '_YYYY_MM_DD_hh_mm_ss'
     """
     return re.sub(r"\_\d{4}(\_\d{2}){5}", "", title)
+
+
+def __retrieve_program_module(test_folder_file_name: Path) -> ModuleType:
+    """Retrives the program module"""
+    items = str(test_folder_file_name).split("/")
+    program = items[items.index(TEST_COLLECTIONS) + 1]
+    module = importlib.import_module(
+        f"{TEST_COLLECTIONS}.{program}.{TEST_ENVIRONMENT_CONFIG_MODULE}"
+    )
+    return module
+
+
+def __retrieve_program_class(test_folder_file_name: Path) -> str:
+    """Looking the a class, inside the given path, that extends TestEnvironmentConfig"""
+    with open(test_folder_file_name, "r") as python_file:
+        parsed_python_file = ast.parse(python_file.read())
+
+        classes = [
+            c
+            for c in parsed_python_file.body
+            if isinstance(c, ast.ClassDef)
+            and any(
+                b
+                for b in c.bases
+                if isinstance(b, ast.Name)
+                and b.id == TEST_ENVIRONMENT_CONFIG_BASE_CLASS_NAME
+            )
+        ]
+
+    # It should have only one occurrence for a class that extends TestEnvironmentConfig
+    return classes[0].name
+
+
+def __retrieve_program_conf() -> Any:
+    PROJECT_ROOT = Path(__file__).parent.parents[1]
+
+    test_collection_folder = os.listdir(PROJECT_ROOT / TEST_COLLECTIONS)
+
+    # Iterate through the folders inside test_collections in order to find the first
+    # occurency for the default_project.config file
+    for program_folder in test_collection_folder:
+        test_folder_file_name = (
+            PROJECT_ROOT
+            / TEST_COLLECTIONS
+            / program_folder
+            / TEST_ENVIRONMENT_CONFIG_PYTHON
+        )
+        # Currently, only one program is supported, so it should consider the first
+        # occurency for default_project.config file
+        if test_folder_file_name.is_file():
+            ProgramConfigClassReference = getattr(
+                __retrieve_program_module(test_folder_file_name),
+                __retrieve_program_class(test_folder_file_name),
+            )
+            default_config_file = (
+                PROJECT_ROOT
+                / TEST_COLLECTIONS
+                / program_folder
+                / TEST_ENVIRONMENT_CONFIG_NAME
+            )
```

Comment:
Should these variables be declared outside the loop? What happens if this line is never executed? Will the return statements fail because they're trying to access variables that don't exist?